dev/tasks/gandiva/travis.linux.yml

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+os: linux
+dist: trusty
+sudo: required
+language: cpp
+services:
+  - docker
+
+# don't build twice
+if: tag IS blank
+
+env:
+  global:
+    - PLAT=x86_64
+    - TRAVIS_TAG={{ task.tag }}
+
+before_install:
+  - eval "${MATRIX_EVAL}"
+  - ulimit -c unlimited -S
+  - |
+      sudo bash -c "echo -e 'Acquire::Retries 10; Acquire::http::Timeout \"20\";' > /etc/apt/apt.conf.d/99-travis-retry"
+      sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+      sudo apt-get update -qq
+matrix:
+  fast_finish: true
+  include:
+  - compiler: gcc
+    language: cpp
+    os: linux
+    jdk: openjdk8
+    env:
+    - ARROW_TRAVIS_USE_TOOLCHAIN=1
+    - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
+    before_script:
+    - git clone -b {{ arrow.branch }} {{ arrow.remote }} arrow
+    - git -C arrow checkout {{ arrow.head }}
+    - mkdir -p dist
+    - arrow/ci/travis_install_linux.sh
+    - arrow/ci/travis_install_clang_tools.sh
+    - export TRAVIS_BUILD_DIR=$TRAVIS_BUILD_DIR/arrow
+    - arrow/ci/travis_install_toolchain.sh
```

Comment:
It will install dependencies via conda. Shouldn't the jar be self-contained?

Reply:
the jar is self contained, we statically link all of the required dependencies and package that in the jar.

are you saying we checkout code and build them instead?

Reply:
Oh cool. However We need a way to ensure that these are packaged/linked correctly.

Reply:
yes absolutely. The tests will ensure that no?

Reply:
I hope so.